webapp-django/crashstats/signature/static/signature/js/signature_report.js

```diff
@@ -1,54 +1,109 @@
 /* global socorro:true, $:true */
 
-$(function () {
+window.SignatureReport = {
+
+    // Function to help with inheritance.
+    'inherit': function (proto) {
+        var f = function () {};
+        f.prototype = proto;
+        return new f ();
+    }
+
+};
+
+SignatureReport.init = function () {
     'use strict';
 
     // parameters
     var searchSection = $('#search-form');
     var form = $('form', searchSection);
+    SignatureReport.form = form;
     var fieldsURL = form.data('fields-url');
     var SIGNATURE = form.data('signature');
     var panelsNavSection = $('#panels-nav');
+    var mainBodyElt = $('#mainbody');
+
+    SignatureReport.pageNum = 1;  // the page number as passed in the URL
+
+    // Define the tab names.
+    var tabNames = ['reports', 'aggregations', 'comments', 'graph'];
+    var tabs = {};
+    var tabClasses = {
+        'reports': SignatureReport.ReportsTab,
+        'aggregations': SignatureReport.AggregationsTab,
+        'comments': SignatureReport.CommentsTab,
+        'graph': SignatureReport.GraphTab
+    };
 
-    var pageNum = 1;  // the page number as passed in the URL
-
-    var initializedTabs = {};
-    var tabsLoadFunctions = {};
+    // Set the current tab, either from location.hash or defaultTab.
+    var defaultTab = tabNames[0];
+    var hashString = window.location.hash.substring(1);
+    var currentTab = hashString ? hashString : defaultTab;
 
-    function getParamsWithSignature() {
+    // Helper function for getting parameters.
+    SignatureReport.getParamsWithSignature = function () {
         var params = form.dynamicForm('getParams');
         params.signature = SIGNATURE;
         return params;
-    }
+    };
+
+    // Helper function for capitalising headings.
+    SignatureReport.capitaliseHeading = function (heading) {
```

Comment:
OK I'll change it :'(

(Apparently it's the fault of the French anyway... https://en.wiktionary.org/wiki/-ise#Usage_notes)